### DISCUSSION TITLE
1.0

### DISCUSSION THREAD

**Discussion by john-livelyvideo** [esfx/esfx#31] [General]
The suspense is killing me, when is 1.0 GA?

**Comment by rbuckton** [comment]
There are a few things I am ironing out currently. I'm doing some general cleanup on the various packages as well as some inlining of "internal" dependencies to cut down on distribution size. There are also several packages that I don't believe are ready for 1.0 just yet (`decorators`, `decorators-stage1-core`, `indexed-object`, `interval`, `reflect-metadata-compat`, and `struct-type`), but lerna doesn't make it easy to version some packages in the monorepo independently. Finally, I'm hoping to improve the performance of `collections-hashmap` and `collections-hashset`, as they are currently significantly slower than native `Map`/`Set` by an order of magnitude. Most of that cost is in the hashing algorithm, so I've been looking into moving that into native code (at least while running in NodeJS).

**Reply by rbuckton** [reply]
On a positive note, I made a change over the weekend that will allow `CancelToken` (from `@esfx/canceltoken`) to be used in place of `AbortSignal` in DOM APIs w/o requiring a shim or adapter.

**Comment by john-livelyvideo** [comment]
@rbuckton I've heard people make arguments for three levels of meta in the past.  Meta on meta on meta, lol.  Needless to say, w/ my tiny brain, can't say I thoroughly understood the argument.  Looking at tools like jetbrains MPS, and the idea of generating multiple levels, is there an example showing decorators and reflect doing this sort of thing?
